.circleci/config.yml

```
@@ -44,3 +44,45 @@ jobs:
           name: Test and coverage report
           command: |
             opam config exec -- make coverage TOKEN=$COVERALLS_REPO_TOKEN
+
+  # This build step ensures we can build a mirage unikernel but doesn't make any attempt to run it
+  # We're basically just trying to ensure our build deps are ok
+  build_mirage:
+    docker:
+      - image: ocaml/opam2:4.09
+        environment:
+          TERM: xterm
+    steps:
+      - checkout
+      - run:
+          name: Update opam
+          command: |
+            opam remote remove default --all
+            opam remote add default https://opam.ocaml.org
+      - run:
+          name: Pin packages
+          command: |
+            for p in pgx pgx_lwt; do
+              opam pin add -y -n ${p%.opam} .
+            done
+      - run:
+          name: Install OCaml dependencies
+          command: opam install --deps-only -y pgx pgx_lwt
+      - run:
+          name: Install mirage
+          command: opam install -y mirage
+      - run:
+          name: Configure mirage
+          command: cd mirage && mirage configure -t xen
```

i think we are on a new enough linux version here where it'll be easier to rely on one of the kvm based backends for mirage. I've had some issue working with xen.

```suggestion
          command: cd mirage && mirage configure -t spt
```

---

What is spt anyway?

---

Mirage supports unikernels that can run on linux/kvm hypervisors via the solo5 project. spt is one available target (https://github.com/Solo5/solo5/blob/master/docs/building.md#spt-running-on-linux-with-a-strict-seccomp-sandbox) 
My understanding is that it uses the linux seccomp to provide sandboxing, but a lot of it is beyond what i understand at the moment :)

---

As long as this target will do the tests we want (ensure we don't depend on any Unix API's), I'm happy.

---

This should indeed be enough to validate that something works with mirage.